server/src/controllers/auth.js

```diff
@@ -1,29 +1,32 @@
+/* eslint-disable no-underscore-dangle */
 import axios from 'axios';
+import jwt from 'jsonwebtoken';
 import User, { validateUser } from '../models/user';
 import Config from '../config/config';
 
+export const generateJWT = (userId) => {
+  const { JWT_SECRET } = Config.server;
+  const token = jwt.sign({ id: userId }, JWT_SECRET); // TODO: add expiry time.
+  return token;
+};
+
 export const checkUser = async (userData) => {
   const { error } = validateUser(userData);
   if (error) {
     return error.details[0].message;
   }
   let user = await User.findOne({ email: userData.email });
   if (user) {
-    return true;
+    return user._id;
   }
   user = new User(userData);
-  user = await user.save();
+  await user.save();
   if (user) {
-    return true;
+    return user._id;
   }
   return false;
```

Comment:
For some reason false does not seem like correct response.
true -or> false is better
or in this case
id -or> null fits better